src/Combinatorics/SimplicialComplexes.jl

```diff
@@ -0,0 +1,353 @@
+using Markdown
+import Oscar: Polymake, pm_object
+
+export
+    SimplicialComplex,
+    betti_numbers,
+    dim,
+    euler_characteristic,
+    facets,
+    f_vector,
+    h_vector,
+    minimal_nonfaces,
+    nvertices,
+    stanley_reisner_ideal,
+    stanley_reisner_ring,
+    load_simplicialcomplex,
+    save_simplicialcomplex,
+    complex_projective_plane,
+    real_projective_plane,
+    klein_bottle,
+    torus # requires a distinction from, e.g., an algebraic group
+
+################################################################################
+##  Constructing
+################################################################################
+
+struct SimplicialComplex
+    pm_simplicialcomplex::Polymake.BigObject
+end
+
+pm_object(K::SimplicialComplex) = K.pm_simplicialcomplex
+
+
+@doc Markdown.doc"""
+    SimplicialComplex(generators::Vector{Vector{Int}})
+
+Construct an abstract simplicial complex from a set of faces.
+While arbitrary nonnegative integers are allowed as vertices, they will be relabeled to consecutive integers starting at 1.
+
+# Example
+```jldoctest
+julia> K = SimplicialComplex([[1,2,3],[2,3,4]])
+Abstract simplicial complex of dimension 2 on 4 vertices
+```
+# Simplicial complex comprising the empty set only
+```jldoctest
+julia> empty = SimplicialComplex(Vector{Set{Int}}([]))
+Abstract simplicial complex of dimension -1 on 0 vertices
+```
+# Example with relabeling
+The original vertices can be recovered.
+```jldoctest
+julia> L = SimplicialComplex([[0,2,17],[2,17,90]]);
+
+julia> facets(L)
+2-element Vector{Vector{Int64}}:
+ [1, 3, 2]
+ [3, 4, 2]
+
+julia> Oscar._vertexindices(Oscar.pm_object(L))
+4-element Vector{Int64}:
+  0
+  2
+ 17
+ 90
+```
+"""
+function SimplicialComplex(generators::Vector{Vector{Int}})
+    K = Polymake.topaz.SimplicialComplex(INPUT_FACES=generators)
+    SimplicialComplex(K)
+end
+
+function SimplicialComplex(generators::Vector{Set{Int}})
```

Comment:
This method looks exactly the same as the previous one. When we have `generators::Union{Vector{Vector{Int}}, Vector{Set{Int}}}` this should be applicable to both desired types.

Also (but this is not important for this draft) we might want to extend these using abstract types instead, e.g. `AbstractVector{<:AbstractVector{<:Base.Integer}}`. The current types suffice right now and with these abstract types you would also have to ensure proper conversion to `Polymake.jl` types. In polyhedral geometry we use `matrix_for_polymake` for matrices. But for these types here we have no existing helper yet.